### PR TITLE
changed onStartCommand with intent to START_REDELIVER_INTENT, added error for null intent (Testing needed)

### PR DESCRIPTION
**Pull request by NightMadness**
Hi Team,
below possibly fixes https://github.com/farkam135/GoIV/issues/477 (to address #535), Testing is need, can someone reproduce the bug?

If we have an null intent we throw an error early so we don't need to dig to the log, conversely we can handle nulls for on destroy.

### CODE REVIEW

**Comment by nahojjjen** [comment]
I haven't been involved enough with this part of the code to make a good review, I think blaisorblade handled this? I'm not sure.

**Comment by Blaisorblade** [comment]
Just noticed this—it fits with what @sarav proposed in https://github.com/farkam135/GoIV/issues/477#issuecomment-249744984 and IIRC the idea made sense. So thanks @NightMadness for the PR. BTW, I've seen you've been rather active recently with good submissions, thanks for that too!

To exercise the change, I'd recommend adding a log statement in `onStartCommand` (temporarily) and try loading other apps to fill memory, to see what happens when GoIV is brought out of memory. Specifically, to have a chance of reproducing #477, I'd try 
- start the service
- get it killed by opening lots of apps
- close the other apps, and hope Android restarts the service
- shut it down via the button in MainActivity

You might even want to try this before the patch (to reproduce the crash) and after (to confirm it's gone), because I'm not actually sure that sequence of events is the correct one.

But overall, unless testing finds some regression, I'd tend to merging this.

**Comment by NightMadness** [comment]
I was able to repreduce while causing a crash in pokefly then clickin on stop on the notification, which is not 100% what we want to test but is close. 

Let me know if you are able to repreduce the bug.

Which reminds me stop in notification should probably changed to skip oncreate and ondestory when executing intent.

**Comment by Blaisorblade** [comment]
> Which reminds me stop in notification should probably changed to skip oncreate and ondestory when executing intent.

I'd guess it shouldn't and can't skip `Pokefly.onDestroy`. Is it running `Pokefly.onCreate`?

**Comment by NightMadness** [comment]
### tl;dr
This pr should handle low memory cases better
We need a separate PR/Bug to handle the "stop" notification bug

--------------
if we have low memory android might try try to start the service then go to 
- **onStartCommand** --> **Null intent** (skipped initialization)
- **stop service** --> **onDestory** --> we get an error since we didn't initialize the view.

I'm not sure where onCreate comes in the cycle...

### one way to reproduce a similar the bug using notification
1. Crash the PokeFly service (any bad code will do).
2. Go to the notification and click on "Stop".
3. We see a new crash.

### to fix it 
1. We change to START_REDELIVER_INTENT (this PR) for restart with low memory cases 
2. stop notification is a different issue could be fixed with a new PR/bug.

**Comment by Blaisorblade** [comment]
I think merging this PR is better than not. On this branch I've renamed quite a few pokemons, tried loading other apps (I think without getting GoIV unloaded), without any problem.

An issue about the extra crash might be good (with the stacktrace), but if the extra crash happens only after a crash that's probably *not* the worst thing (it's not even clear whether we want to fix that). I see the goal was to test what happens if the service is closed instead... but I guess, after this PR, the remaining bug only happens after a crash.
